src/Nethermind/Nethermind.Facade/BlockchainBridge.cs

```diff
@@ -191,7 +194,39 @@ public CallOutput EstimateGas(BlockHeader header, Transaction tx, CancellationTo
                 estimateGasTracer.WithCancellation(cancellationToken));
             
             long estimate = estimateGasTracer.CalculateEstimate(tx);
-            return new CallOutput {Error = tryCallResult.Success ? estimateGasTracer.Error : tryCallResult.Error, GasSpent = estimate, InputError = !tryCallResult.Success};
+            
+            return new CallOutput 
+            {
+                Error = tryCallResult.Success ? estimateGasTracer.Error : tryCallResult.Error, 
+                GasSpent = estimate, 
+                InputError = !tryCallResult.Success
+            };
+        }
+
+        public CallOutput CreateAccessList(BlockHeader header, Transaction tx, CancellationToken cancellationToken, bool optimize)
+        {
+            CallOutputTracer callOutputTracer = new();
+            AccessTxTracer accessTxTracer = optimize 
+                ? new(tx.SenderAddress, 
+                    tx.GetRecipient(tx.IsContractCreation ? _stateReader.GetNonce(header.StateRoot, tx.SenderAddress) : 0)) 
+                : new();
+
+            (bool Success, string Error) tryCallResult = TryCallAndRestore(header, header.Number, header.Timestamp, tx,
+                new CancellationTxTracer(new CompositeTxTracer(callOutputTracer, accessTxTracer), cancellationToken)
+                {
+                    IsTracingActions = true, 
```

Comment:
when would you ever set it to false on CancellationTracer? can we simply make it hardcoded true there?

Reply:
We don't want to trace on some things that are called too frequent like Instructions, due to performance.